src/postgres_deparse.c

```diff
@@ -246,89 +251,144 @@ static void deparseAnyNameSkipLast(StringInfo str, List *parts)
 	}
 }
 
-// "a_expr" / "b_expr" in gram.y
-static void deparseExpr(StringInfo str, Node *node)
```

Comment:
This refactoring maps the deparser handling of a/b/c type expressions closer to how `gram.y` is written, and helps us only add parenthesis around `a_expr` type expressions when needed (i.e. when they are used within the context of a `c_expr`).